docs/proposals/buildstrategy-annotation-propagation.md

```diff
@@ -0,0 +1,165 @@
+
+> A slim BuildStrategy is one where the BuildStrategy author gets to accomplish more by specifying less.
+
+But, beside the steps that are to be performed, the build strategy administrator already CAN define certain runtime behavior aspects like container resources and security context.
+
+By enabling the administrator to also control the annotations of the build run pod IF necessary, more scenarios CAN be supported.
+
+## Motivation
+
+The runtime behavior of a pod is not only specified in the containers. There are use cases where behavior of pods is described through annotations. This is especially the case for alpha and beta features of Kubernetes. Examples:
+
+- The Kubernetes [Network Traffic Shaping](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/#support-traffic-shaping) feature looks for the `kubernetes.io/ingress-bandwidth` and `kubernetes.io/egress-bandwidth` annotations to limit the network bandwidth the pod is allowed to use.
+- The [SecComp profile selection](https://kubernetes.io/docs/tutorials/clusters/seccomp/#create-a-pod-with-a-seccomp-profile-for-syscall-auditing) used to be done through the `seccomp.security.alpha.kubernetes.io/pod` annotation until Kubernetes 1.18 - 1.19 makes it a first-class property in the Pod's security context.
+- The [AppArmor profile of a container](https://kubernetes.io/docs/tutorials/clusters/apparmor/) is defined using the `container.apparmor.security.beta.kubernetes.io/<container_name>` annotation.
+
+To use those features for build run pods, control over the pod's annotations is necessary. As those features are clearly something that administrators should define rather then the users that define builds and build runs, it makes sense to restrict this feature to the (cluster) build strategy only.
+
+### Goals
+
+- Enable the build strategy administrator to define annotations that are copied to the build run's pod.
+
+### Non-Goals
+
+- Enable users to define annotations on build and build run that are copied to the build run's pod.
+- Enable anybody to define labels on one of our CRDs and make them appear on the build run's pod. This should be covered elsewhere.
+
+## Proposal
+
+In the (cluster) build strategy, the build strategy administrator can define annotations in the metadata. This is possible in the same way as for all Kubernetes objects, see the [Annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/) topic in the Kubernetes documentation. Annotations are key/value pairs. The key consists of up to two parts: an optional prefix and a name. If both are defined, the `/` is used as separators. The prefix must be a DNS subdomain. Kubernetes reserves the `kubernetes.io` and `k8s.io` prefixes for its own core components. In build we already use annotations in the build controller using the `build.build.dev` prefix. Two use cases:
```

Comment:
I am not sure if we need to mention cluster here: `In the (cluster) build strategy`

I think it works for both cluster scope buildstrategy and namespace scope buildstrategy. So I think we just need to mention `In the build strategy` to avoid confusion.

Reply:
Changed to

> In the `BuildStrategy` and `ClusterBuildStrategy`

to make it consistent with the other requested change (use real CRD name).